lib/mp4/transmuxer.js

```diff
@@ -215,14 +215,16 @@ AudioSegmentStream = function(track) {
     }
 
     baseMediaDecodeTimeTs = clock.audioTsToVideoTs(track.baseMediaDecodeTime, track.samplerate);
+    // determine frame clock duration based on sample rate, round up to avoid overfills
+    frameDuration = Math.ceil(ONE_SECOND_IN_TS / (track.samplerate / 1024));
 
     if (audioAppendStartTs &&
         videoBaseMediaDecodeTime &&
-        // old audio doesn't overlap with new audio
-        audioAppendStartTs < baseMediaDecodeTimeTs) {
-      audioGapDuration = baseMediaDecodeTimeTs - videoBaseMediaDecodeTime;
-      // determine frame clock duration based on sample rate, round up to avoid overfills
-      frameDuration = Math.ceil(ONE_SECOND_IN_TS / (track.samplerate / 1024));
+        // gap is at least a frame duration
+        baseMediaDecodeTimeTs - audioAppendStartTs >= frameDuration) {
```

Comment:
I feel like this condition should be:

`baseMediaDecodeTimeTs - Math.max(audioAppendStartTs, videoBaseMediaDecodeTime) >= frameDuration` 

...but if that is the case then we can just remove the condition completely since it'll be calculated below and, if this were to evaluate to false, would result in a value that is `audioFillFrameCount < 1` and we will exit early.